File: dxbase/src/main/java/com/dxmovie/dxbase/binding/viewadapter/checkbox/ViewAdapter.java
package com.dxmovie.dxbase.binding.viewadapter.checkbox;

import android.widget.CheckBox;
import android.widget.CompoundButton;

import androidx.databinding.BindingAdapter;

import com.dxmovie.dxbase.binding.command.BindingCommand;
import com.dxmovie.dxbase.binding.command.BindingCommandWithParams;

/**
 */

public class ViewAdapter {
    /**
     * @param bindingCommand //绑定监听
     */
    @SuppressWarnings("unchecked")
    @BindingAdapter(value = {"onCheckedChangedCommand"}, requireAll = false)
    public static void setCheckedChanged(final CheckBox checkBox, final BindingCommandWithParams<Boolean> bindingCommand) {
        checkBox.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(CompoundButton compoundButton, boolean b) {
                bindingCommand.executeAction(b);
            }
        });
    }
}

File: dxbase/src/main/java/com/dxmovie/dxbase/binding/viewadapter/checkbox/ViewAdapter.kt
package com.dxmovie.dxbase.binding.viewadapter.checkbox

import android.widget.CheckBox
import androidx.databinding.BindingAdapter
import com.dxmovie.dxbase.binding.command.BindingCommandWithParams

class ViewAdapter {

    companion object {
        /**
         * @param bindingCommand //绑定监听
         */
        @JvmStatic
        @BindingAdapter(value = ["onCheckedChangedCommand"], requireAll = false)
        fun setCheckedChanged(checkBox: CheckBox, bindingCommand: BindingCommandWithParams<Boolean?>) {
            checkBox.setOnCheckedChangeListener { compoundButton, b -> bindingCommand.executeAction(b) }
        }
    }

}